datafusion/common/src/config.rs

```diff
@@ -1368,6 +1371,10 @@ pub struct TableParquetOptions {
     pub global: ParquetOptions,
     /// Column specific options. Default usage is parquet.XX::column.
     pub column_specific_options: HashMap<String, ColumnOptions>,
+    /// Optional, additional metadata to be inserted into the key_value_metadata
+    /// for the written [`FileMetaData`](https://docs.rs/parquet/latest/parquet/file/metadata/struct.FileMetaData.html).
+    #[cfg(feature = "parquet")]
+    pub key_value_metadata: Option<Vec<KeyValue>>,
```

Comment:
I think Key/Value is just owned strings.  https://docs.rs/parquet/latest/parquet/format/struct.KeyValue.html

So we could avoid the `cfg`s by doing something like
```suggestion
    pub key_value_metadata: HashMap<String, Option<String>>
```

ANd then translating that to KeyValues during the write

This would also make the protobuf easier to handle (just follow the encoding for HashMaps)